### PR TITLE
fix:  heapify() function does not work  in heap_sort.cpp

### PR DESCRIPTION
**Pull request by liushubin-gitHub**
fix bug:  heapify() function does not work, during heap sort;
    solve c++ build error.
@ericcurtin Eric Curtin  
pls review.

### CODE REVIEW

**Comment by ericcurtin** [comment]
I must fix it here too when this gets through:

https://github.com/TheAlgorithms/C/pull/508

**Comment by liushubin-gitHub** [comment]
thank you very much。

File: sorting/heap_sort.cpp [hunk 1]
```diff
@@ -47,6 +47,4 @@ int main() {
     for (int i = 0; i < n; ++i) {
         std::cout << a[i] << std::endl;
     }
-
-    std::getchar();
```
Comment (Panquesito7) [inline sorting/heap_sort.cpp:51]: Why remove this?

Comment (liushubin-gitHub) [reply sorting/heap_sort.cpp:51]: std::getchar() build failed.
so I'm just remove it.

Comment (Panquesito7) [reply sorting/heap_sort.cpp:51]: You can use `std::cin.get()`.

**Comment by kvedala** [comment]
a possible duplicate of #873

**Comment by liushubin-gitHub** [comment]
OK i will check it on the latest version.
To make sure the bug fixed

File: sorting/heap_sort.cpp (hunk 1 above)
Comment (liushubin-gitHub) [reply sorting/heap_sort.cpp:51]: thank you again

**Comment by liushubin-gitHub** [comment]
> a possible duplicate of #873
> 
> @ayaankhan98 and @liushubin-gitHub , referencing the bug issue number in the pull-request descriptions could help us avoid such conflicts. Hence, we have the checkmark in the pull-request description. Please use it and not not delete the template. This pull-request did not use the description template.

=======================================
I checkout the latest  code , and test ok.

**Comment by liushubin-gitHub** [comment]
Since the heap sort works well in the latest code.  (Although the modifications are different from this pull request)
As the bug has fixed , so close it.

**Pull request closed without merging**